irohad/multi_sig_transactions/state/impl/mst_state.cpp

```diff
@@ -93,17 +108,28 @@ namespace iroha {
   }
 
   MstState MstState::eraseByTime(const TimeType &time) {
-    MstState out = MstState::empty(log_, completer_);
+    MstState out = MstState::empty(log_, completer_, txs_limit_);
     while (not index_.empty() and (*completer_)(index_.top(), time)) {
       auto iter = internal_state_.find(index_.top());
 
       out += *iter;
+      assert(txs_amount_ >= boost::size((*iter)->transactions()));
```

Comment:
And what should happen when an assertion fails at runtime?

Reply:
a crash that forces one to investigate the problem. this chack should never fail - it is guaranteed by the code for now. the assertion is needed for later changes, when somebody possibly breaks this contract.